From: Xavier Claessens <xavier.claessens@collabora.com>
Date: Mon, 7 Dec 2020 11:40:20 -0500
Subject: [PATCH] pkgconfig: Respect variable ordering when passed as list

This fix a regression introduced in Meson 0.56.0 when using python 3.5.
Also mention in documentation that using a meson dict does not guarantee
ordering.

Fixes: #8074.
---
 mesonbuild/interpreter.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/interpreter.py b/mesonbuild/interpreter.py
index 6222f97a71d8..b0cbecc75114 100644
--- a/mesonbuild/interpreter.py
+++ b/mesonbuild/interpreter.py
@@ -2690,7 +2690,7 @@ class Interpreter(InterpreterBase):
             varlist = mesonlib.stringlistify(variables)
             if list_new:
                 FeatureNew.single_use('variables as list of strings', '0.56.0', self.subproject)
-            variables = {}
+            variables = collections.OrderedDict()
             for v in varlist:
                 try:
                     (key, value) = v.split('=', 1)
